Hunted Mobile/Hunted Mobile/Hunted Mobile/Model/Map.cs

```diff
@@ -1,69 +1,66 @@
 
-using Mapsui.Geometries;
-using Mapsui.UI.Forms;
-using Mapsui.UI.Objects;
-
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Linq;
 
 namespace Hunted_Mobile.Model {
     public class Map {
-        private List<Player> users = new List<Player>();
-
-        private List<Loot> loot = new List<Loot>();
-
-        public Player PlayingUser { get; set; }
-        public Boundary GameBoundary { get; set; }
+        private readonly ObservableCollection<Player> players = new ObservableCollection<Player>();
+        private readonly ObservableCollection<Loot> loot = new ObservableCollection<Loot>();
 
-        public List<Police> Police => users.Where(user => user is Police).Select(user => new Police(user)).ToList();
-        public List<Thief> Thiefs => users.Where(user => user is Thief).Select(user => new Thief(user)).ToList();
+        public ICollection<Player> Players {
+            get => players;
+            set {
+                players.Clear();
 
-        public Map() { }
-
-        public void AddUser(Player user) {
-            if(user.IsValid) {
-                users.Add(user);
+                foreach(var item in value) {
+                    players.Add(item);
+                }
             }
         }
-
-        public void RemoveUser(Player user) {
-            users.Remove(user);
+        public ICollection<Loot> Loot {
+            get => loot;
+            set {
+                loot.Clear();
+
+                foreach(var item in value) {
+                    loot.Add(item);
+                }
+            }
         }
 
-        public IEnumerable<Player> GetUsers() {
-            return users.AsReadOnly();
-        }
+        public IReadOnlyCollection<Police> Police => Players.Where(user => user is Police).Select(user => new Police(user)).ToList();
+        public IReadOnlyCollection<Thief> Thiefs => Players.Where(user => user is Thief).Select(user => new Thief(user)).ToList();
 
-        public Player GetUserById(int id) {
-            return users.FirstOrDefault(user => user.Id == id);
-        }
+        public Player PlayingUser { get; set; }
+        public Boundary GameBoundary { get; set; }
 
-        public void SetUsers(IEnumerable<Player> users) {
-            this.users = users.Where(user => user.IsValid).ToList();
+        public Map() {
+            players.CollectionChanged += CollectionChanged;
+            loot.CollectionChanged += CollectionChanged;
         }
 
-        public void AddLoot(Loot loot) {
-            if(loot.IsValid) {
-                this.loot.Add(loot);
+        private void CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) {
+            if(e.NewItems != null) {
+                foreach(var item in e.NewItems) {
+                    if(item is Player) {
+                        if(!((Player) item).IsValid) {
+                            e.NewItems.Remove(item);
```

Comment:
Gaat dat wel goed als er op de array wordt geloopt, maar ook tegelijkertijd items worden verwijderd?

Reply:
Nee, `NewItems` is zelfs een ReadOnlyList, maar ik wilde afdwingen dat er alleen valide spelers een loot toegevoegd kunnen worden. Dit geeft een OperationNotSupportedException maar er is anders geen manier om de 'change' te annuleren. 
Maar misschien is een `throw ...` beter.